client/src/components/User/ExternalIdentities/ExternalLogin.vue

```diff
@@ -271,7 +271,12 @@ function getIdpPreference() {
                         <BButton class="d-block mt-3" @click="submitOIDCLogin(idp)">
                             <i :class="oIDCIdps[idp]" />
                             Sign in with
-                            {{ idp.charAt(0).toUpperCase() + idp.slice(1) }}
+                            <span v-if="iDPInfo['label']">
+                                {{ iDPInfo["label"].charAt(0).toUpperCase() + iDPInfo["label"].slice(1) }}
+                            </span>
+                            <span v-else>
+                                {{ idp.charAt(0).toUpperCase() + idp.slice(1) }}
+                            </span>
```

Comment:
Just a minor refactoring improvement. You could move this `string.charAt(0).toUpperCase() + string.slice(1)` to a function `toTitle` or similar under `client/src/utils/strings.ts` to improve reusability and readability.

Reply:
Done, thank you!